### ISSUE TITLE
realm::TableViewBase::is_in_sync() const - endless loop crash on deleteAllObjects

### ISSUE THREAD

**Issue by genadyo** [realm/realm-cocoa#2925]
This is basically the code I'm running:    

```objective-c
RLMRealm *realm = [RLMRealm defaultRealm];
[realm beginWriteTransaction];
[realm deleteAllObjects];
[realm commitWriteTransaction];
```

Maybe it happens on the write after deleting all.

Here is the crash:
http://crashes.to/s/7ec302d5c8f

**Comment by mrackwitz** [comment]
Thanks for reporting that. Can you safely reproduce this issue?
Could you share a bit more about the schema and possible also the object graph you had in your Realm before you deleted all objects? If you have a Realm file with which you can reproduce this issue and could send it to us (help@realm.io), this would be super helpful.

**Comment by genadyo** [comment]
I think that I was still holding strong pointers to some of the realm objects and that was the cause of that crash, does it make sense? I fixed it and I don't see it anymore.

**Comment by mrackwitz** [comment]
It could be very likely related to that, but even in that case all existing accessors should be just invalidated and become inaccessible. So you definitively have to clear up after deleting all objects, but it shouldn't crash. You would hit an assert though if you would actually access one of the remaining accessors.

**Comment by mrackwitz** [comment]
The latest release brings a new version of the underlying storage engine. Can you re-test this please and would report whether this issue still occurs?

**Comment by jpsim** [comment]
As @mrackwitz said, we believe this was fixed by #2954, which is available in `master`. Could you please integrate that version of Realm into your app and reopen this issue if this occurs again?
